src/build_options.rs

```diff
@@ -560,6 +564,15 @@ impl BuildOptions {
         let generate_import_lib = is_generating_import_lib(&cargo_metadata)?;
         let interpreter = if self.find_interpreter {
             // Auto-detect interpreters
+            let requires_python = metadata23.requires_python.as_ref();
+            if requires_python.is_some() {
+                let versions = requires_python.unwrap();
+                let has_pre_release = versions.iter().any(|v| v.any_prerelease());
+                if has_pre_release && !self.allow_prereleases {
+                    bail!("⚠️  Warning: python version is pre release, need pass flag --allow-prereleases");
```

Comment:
I think we should just skip it and print the warning if `allow_prereleases` isn't set, no need to bail.

Reply:
ok

Reply:
code changed, but i not sure what i do is right, i not familiar with rust.